doc/case-studies/timelock/env.tla

```diff
@@ -0,0 +1,83 @@
+(*
+ * Model of Soroban environment
+ * 
+ * Andrey Kuprianov, 2024
+ *)
+
+---- MODULE env ----
+
+EXTENDS Integers, Sequences, Apalache
+
+(*
+    @typeAlias: address = Str;
+    @typeAlias: dataKey = Str;
+*)
+typedefs == TRUE
+
+VARIABLES
+    (*  Soroban's environment 
+        @type: {
+            current_contract_address: $address,
+            ledger_timestamp: Int
+        };
```

Comment:
```suggestion
        @type: {
            current_contract_address: $address,
            timestamp: Int,
            sequence: Int,
            invoked_function_name: Str
        };
```
I'd recommend making this the `env` type alias to use for `env`-method parameters.